### PR TITLE
Moves warning about multiple UARTs with RC protocol to a pre-arm so its visible to users

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

**Comment by Hwurzburg** [comment]
tested in SITL

File: libraries/AP_SerialManager/AP_SerialManager.cpp [hunk 1]
```diff
@@ -543,11 +543,7 @@ void AP_SerialManager::init()
 
 #ifndef HAL_BUILD_AP_PERIPH
                 case SerialProtocol_RCIN:
-                    if (AP::RC().has_uart()) {
-                        GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Duplicate RCIN configured on SERIAL%u", i);
-                    } else {
-                        AP::RC().add_uart(uart);
-                    }
+                    AP::RC().add_uart(uart);
```
Comment (IamPete1) [inline libraries/AP_SerialManager/AP_SerialManager.cpp:546]:
```suggestion
                    if (!AP::RC().has_uart()) {
                        AP::RC().add_uart(uart);
                    }
```
Just to maintain current behaviour.

File: libraries/AP_Arming/AP_Arming.cpp [hunk 2]
```diff
@@ -1411,6 +1412,16 @@ bool AP_Arming::opendroneid_checks(bool display_failure)
     return true;
 }
 
+//Check for multiple RC in serial protocols
+bool AP_Arming::serial_protocol_checks(bool display_failure)
+{
+    if (AP::serialmanager().have_serial(AP_SerialManager::SerialProtocol_RCIN, 1)) {
+       check_failed(display_failure, "Multiple UARTs configured for RC input");
```
Comment (IamPete1) [inline libraries/AP_Arming/AP_Arming.cpp:1419]:
```suggestion
       check_failed(display_failure, "Multiple serial ports configured for RC input");
```
Serial better aligns with the param names for serial functions .

**Review by andyp1per — CHANGES_REQUESTED** [review]
@IamPete1 's comments are correct

File: libraries/AP_SerialManager/AP_SerialManager.cpp (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_SerialManager/AP_SerialManager.cpp:546]: right

**Review by tridge — APPROVED** [review]

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**